src/coreComponents/managers/Functions/TableFunction.cpp

```diff
@@ -165,8 +188,10 @@ void TableFunction::reInitializeFunction()
     increment *= m_size[ii];
   }
 
-  // Error checking
-  GEOSX_ERROR_IF( increment != m_values.size(), "Table dimensions do not match!" );
+  if( m_coordinates.size() > 0 && m_values.size() > 0 ) // coordinates and values have been set
+  {
+    GEOSX_ERROR_IF( increment != m_values.size(), "Table dimensions do not match!" );
```

Comment:
Is there a check in the original/new version to make sure that the axes are monotonic (maybe I'm missing it)?  That could certainly cause some unclear error messages...

Reply:
@cssherman Good idea, there was not check in `TableFunction` originally, but I added one in the relperm constitutive model that uses the tables. In the latest commit I moved it to `TableFunction`, so the check will be applied for all the tables, not all relperm tables. Thanks!